registry/remote/registry_test.go

```diff
@@ -194,3 +194,45 @@ func TestRegistry_Repository(t *testing.T) {
 		t.Errorf("Registry.Repository() = %v, want %v", got, want)
 	}
 }
+
+func testRegistry() *httptest.Server {
+	return httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		result := struct {
+			Repositories []string `json:"repositories"`
+		}{
+			Repositories: []string{"public/repo1", "public/repo2", "internal/repo3"},
+		}
+		json.NewEncoder(w).Encode(result)
+	}))
+}
+
+// This is a example for listing respositories in the registry:
+func ExampleRegistry_Repositories() {
+	// Mocking local registry
+	ts := testRegistry()
+	defer ts.Close()
+	uri, err := url.Parse(ts.URL)
+	if err != nil {
+		panic(err)
+	}
+
+	// Example: List repositories in a registry
+	reg, err := NewRegistry(uri.Host) // Create a registry via the remote host
+	if err != nil {
+		panic(err) // Handle error
+	}
+	reg.RepositoryOptions.PlainHTTP = true // Use HTTP
+	fn := func(repos []string) error {     // Setup a callback function to process returned repository list
+		for _, repo := range repos {
+			fmt.Println(repo)
+		}
+		return nil
+	}
+
+	ctx := context.Background()
+	reg.Repositories(ctx, fn)
```

Comment:
You may also want to show something like this first:

```go
ctx := context.Background()
repos, err := registry.Repositories(ctx, reg)
if err != nil {
    panic(err)
}
fmt.Println(repos)
```

Reply:
It should be added to another function `ExampleRepositories`, will do that